prometheus/histogram.go

```diff
@@ -1102,6 +1129,10 @@ func (h *histogram) updateExemplar(v float64, bucket int, l Labels) {
 		panic(err)
 	}
 	h.exemplars[bucket].Store(e)
+	doSparse := h.nativeHistogramSchema > math.MinInt32 && !math.IsNaN(v)
+	if doSparse {
+		h.nativeExemplars.addExemplar(e)
+	}
```

Comment:
The doc comment for this method (lines 1121-1122) needs an update to mention that nativeExemplars are handled, too.